github/git_commits.go

```diff
@@ -88,29 +88,33 @@ type createCommit struct {
 }
 
 // CreateCommit creates a new commit in a repository.
+// commit must not be nil.
 //
 // The commit.Committer is optional and will be filled with the commit.Author
 // data if omitted. If the commit.Author is omitted, it will be filled in with
 // the authenticated user’s information and the current date.
 //
 // GitHub API docs: https://developer.github.com/v3/git/commits/#create-a-commit
 func (s *GitService) CreateCommit(ctx context.Context, owner string, repo string, commit *Commit) (*Commit, *Response, error) {
+	if commit == nil {
+		return nil, nil, fmt.Errorf("commit must be provided")
```

Comment:
Given that this func (and `PullRequestsService.Edit`) is now documented that "commit must not be nil", and it returns an error, we have two choices of what to do if the user uses the API incorrectly and `commit` is actually `nil`:

1. Panic.
2. Return an error (currently being done).

Which of those two options do you think is better @gmlewis any why?